packages/grid/x-data-grid/src/components/panel/GridColumnsPanel.tsx

```diff
@@ -163,12 +172,23 @@ function GridColumnsPanel(props: GridColumnsPanelProps) {
   );
 
   const currentColumns = React.useMemo(() => {
+    const togglableColumns = getTogglableColumns ? getTogglableColumns(sortedColumns) : null;
```

Comment:
Did you consider treating `getTogglableColumns` as a filter function with the same return type as the input?
```ts
getTogglableColumns?: (columns: GridColDef[]) => GridColDef['field'][];
```
Then we could use the value returned by `getTogglableColumns` instead of `sortedColumns`.

What do you think?

Reply:
That's a good point but actually we [decided](https://github.com/mui/mui-x/pull/6333#discussion_r986795858) not to follow this approach in `getColumnForNewFilter` and `filterColumns` while working on https://github.com/mui/mui-x/pull/6333, so I tried to be be consistent with it here.

Do you see a drawback by doing it this way?

Reply:
I did not realize we already have a similar prop.
My only concern here is that we unnecessarily map `GridColDef[]` to `GridColDef['field'][]`, and then map it back to `GridColDef[]`.
Consistency with the `filterColumns` prop is the only reason to do it like this.
Decision is up to you :)

Reply:
I would keep it this way for API consistency.

It could also become easier to use if the users want to use it without reusing the `columns` attribute (I imagine this might be simpler for some users):

```tsx
getTogglableColumns = () => ['field1', 'field2'];
```